misk/src/main/kotlin/misk/web/DispatchMechanism.kt

```diff
@@ -19,19 +18,20 @@ enum class DispatchMechanism {
    * This can be misleading: web sockets don't behave like normal GETs (they are upgraded), and GRPC
    * calls don't behave like normal POSTS (they are duplex).
    */
-  val method: HttpMethod
+  val method: String
```

Comment:
We possibly want this as a new enum to avoid comparing strings.

Reply:
do we really need that if `DispatchMechanism` is already an enum? also this value could be a constructor param on this enum instead

Reply:
That's a good point, method is really just a subset of `DispatchMechanism`, so method could probably return a `DispatchMechanism` instead for cases like `WEBSOCKET` and `GET`

Reply:
Why not use HttpMethod?

Reply:
OMG, it’s because HttpMethod doesn’t have PATCH! Craxy

Reply:
Yeah, Jetty stuck a little too close to the spec on that one. Should we keep it as a string? Or try to use an enum?

Reply:
String is fine.